registry/proxy/proxyauth.go

```diff
@@ -33,16 +33,42 @@ func (c credentials) SetRefreshToken(u *url.URL, service, token string) {
 }
 
 // configureAuth stores credentials for challenge responses
-func configureAuth(username, password string) (auth.CredentialStore, error) {
-	creds := map[string]userpass{
-		tokenURL: {
+func configureAuth(username, password, remoteURL string) (auth.CredentialStore, error) {
+	creds := map[string]userpass{}
+	authURLs, err := getAuthURLs(remoteURL)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, url := range authURLs {
+		context.GetLogger(context.Background()).Infof("Discovered token authentication URL: %s", url)
+		creds[url] = userpass{
 			username: username,
 			password: password,
-		},
+		}
 	}
+
 	return credentials{creds: creds}, nil
 }
 
+func getAuthURLs(remoteURL string) ([]string, error) {
+	authURLs := []string{}
+	resp, err := http.Get(remoteURL + "/v2/")
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range auth.ResponseChallenges(resp) {
+		if c.Scheme == "bearer" {
```

Comment:
Can you use `strings.EqualFold` here instead of `==`. The value is returned as `Bearer` and it might get converted to lower somewhere but would prefer to just do a case insensitive check.